filter.cc

```diff
@@ -387,6 +360,44 @@ bool Filter::hasUnsupportedOptions() const {
   return (filterOption & FOUnsupportedSoSkipCheck) != 0;
 }
 
+bool Filter::contextDomainMatchesFilter(const char *contextDomain) {
```

Comment:
This checks if a context domain is applicable to this filter.
For example if `foo.example.com` is the context domain then it will check if this filter is applicable to both `foo.example.com` and `example.com`.